src/pause/140C70.c

```diff
@@ -1,17 +1,244 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "pause/140C70", func_8024D930);
+#define pause_map_tab_places_desc_string_id 0x1D00F2
 
-INCLUDE_ASM(s32, "pause/140C70", pause_map_draw_border_arrows);
+void func_8024D930(PauseMapSpace* space) {
+    Vec2b* path = space->path;
+    s32 pathLength = space->pathLength;
+    s32 x1 = pause_map_spaces[space->parent].xPos;
+    s32 y1 = pause_map_spaces[space->parent].yPos;
+    s32 x2 = space->xPos;
+    s32 y2 = space->yPos;
+    s32 xThingPrev = 0;
+    s32 yThingPrev = 0;
+    s32 i;
+
+    for (i = 0; i < pathLength; i++) {
+        s32 xThing = ((x1 - x2) * (i + 1)) / (pathLength + 1);
+        s32 yThing = ((y1 - y2) * (i + 1)) / (pathLength + 1);
+
+        path[i].x = xThing - xThingPrev;
+        path[i].y = yThing - yThingPrev;
+        xThingPrev = xThing;
+        yThingPrev = yThing;
+    }
+}
+
+void pause_map_draw_border_arrows(s32 imageIndex, s32 x, s32 y) {
+    s32 size = 16;
+
+    switch (imageIndex) {
+        case 0:
+            pause_draw_rect(x * 4, y * 4, (x + size) * 4, (y + size) * 4, 0, 0, 0, 0x400, 0x400);
+            break;
+        case 1:
+            pause_draw_rect(x * 4, y * 4, (x + size) * 4, (y + size) * 4, 0, 0, 0x200, 0x400, 0x400);
+            break;
+        case 2:
+            pause_draw_rect(x * 4, y * 4, (x + size) * 4, (y + size) * 4, 0, 0, 0x400, 0x400, 0x400);
+            break;
+        default:
+            pause_draw_rect(x * 4, y * 4, (x + size) * 4, (y + size) * 4, 0, 0, 0x600, 0x400, 0x400);
+            break;
+    }
+}
 
 INCLUDE_ASM(s32, "pause/140C70", pause_map_draw_contents);
 
-INCLUDE_ASM(s32, "pause/140C70", pause_map_draw_title);
+void pause_map_draw_title(s32* arg1, s32 arg2, s32 textOffsetY, s32 textOffsetX) {
+    s32 stringWidth;
+    s32 stringID;
```

Comment:
Call these msgWidth and msgID; we're calling strings messages in most places except from the not-yet-renamed string funcs.

Reply:
I changed them, but it's a bit weird that all the functions pertaining to these say string and not msg

Reply:
See #413